PxCs/Interface/PxMorphMesh.cs

```diff
@@ -0,0 +1,141 @@
+using System.Numerics;
+using System.Runtime.InteropServices;
+
+namespace PxCs.Interface
+{
+    public class PxMorphMesh
+    {
+        private const string DLLNAME = PxPhoenix.DLLNAME;
+
+        [DllImport(DLLNAME)] public static extern IntPtr pxMmbLoad(IntPtr buffer);
+        [DllImport(DLLNAME)] public static extern IntPtr pxMmbLoadFromVdf(IntPtr vdf, string name);
+        [DllImport(DLLNAME)] public static extern void pxMmbDestroy(IntPtr mmb);
+        [DllImport(DLLNAME)] public static extern IntPtr pxMmbGetName(IntPtr mmb);
+        [DllImport(DLLNAME)] public static extern IntPtr pxMmbGetMesh(IntPtr mmb);
+        [DllImport(DLLNAME)] public static extern uint pxMmbGetMorphPositionCount(IntPtr mmb);
+        [DllImport(DLLNAME)] public static extern Vector3 pxMmbGetMorphPosition(IntPtr mmb, uint i);
+        [DllImport(DLLNAME)] public static extern uint pxMmbGetAnimationCount(IntPtr mmb);
+        [DllImport(DLLNAME)] public static extern IntPtr pxMmbGetAnimation(IntPtr mmb, uint i);
+        [DllImport(DLLNAME)] public static extern IntPtr pxMmbAniGetName(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern int pxMmbAniGetLayer(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern float pxMmbAniGetBlendIn(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern float pxMmbAniGetBlendOut(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern float pxMmbAniGetDuration(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern float pxMmbAniGetSpeed(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern uint pxMmbAniGetFlags(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern uint pxMmbAniGetFrameCount(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern IntPtr pxMmbAniGetVertices(IntPtr ani, out int length);
+        [DllImport(DLLNAME)] public static extern uint pxMmbAniGetSampleCount(IntPtr ani);
+        [DllImport(DLLNAME)] public static extern Vector3 pxMmbAniGetSample(IntPtr ani, uint i);
+
+
+        public static PxMorphMeshData? LoadMorphMeshFromVdf(IntPtr vdfPtr, string name)
+        {
+            var mmbPtr = pxMmbLoadFromVdf(vdfPtr, name);
+            if (mmbPtr == IntPtr.Zero)
+                return null;
+
+            var data = GetMorphMeshFromPtr(mmbPtr);
+            return data;
+
+        }
+
+        public static PxMorphMeshData GetMorphMeshFromPtr(IntPtr morphMeshPtr)
+        {
+            var name = GetName(morphMeshPtr);
+            var mesh = GetMeshData(morphMeshPtr);
+            var positions = GetPositionData(morphMeshPtr);
+            var animations = GetAnimationData(morphMeshPtr);
+            return new PxMorphMeshData()
+            {
+                name = name,
+                mesh = mesh,
+                positions = positions,
+                animations = animations
+            };
+        }
+
+        public static string GetName(IntPtr mmb)
+        {
+            return pxMmbGetName(mmb).MarshalAsString();
```

Comment:
You can remove this method and shorten it up. It's sufficient to just juse pxGetSomething(ptr).MarshalAsString(). --> The extension method is used quite often. I assume every user knows what to with a ptr returned.

Reply:
And done